sdk/eventgrid/azure-eventgrid/tests/eventgrid_preparer.py

```diff
@@ -0,0 +1,93 @@
+import functools
+import hashlib
+import os
+from collections import namedtuple
+
+from azure.mgmt.eventgrid import EventGridManagementClient
+from azure.mgmt.eventgrid.models import Topic, InputSchema
+from azure_devtools.scenario_tests.exceptions import AzureTestError
+
+from devtools_testutils import (
+    ResourceGroupPreparer, AzureMgmtPreparer, FakeResource
+)
+
+from devtools_testutils.resource_testcase import RESOURCE_GROUP_PARAM
+
+EVENTGRID_TOPIC_PARAM = 'eventgrid_topic'
+EVENTGRID_TOPIC_LOCATION = 'westus'
+CLOUD_EVENT_SCHEMA = InputSchema.cloud_event_schema_v1_0
+
+# Shared base class for event grid sub-resources that require a RG to exist.
+class _EventGridChildResourcePreparer(AzureMgmtPreparer):
```

Comment:
are there child resources any more that require this wrapper layer?  it looks like only one preparer.